src/mach/mod.rs

```diff
@@ -313,16 +321,36 @@ impl<'a> Iterator for FatArchIterator<'a> {
     }
 }
 
-/// Iterator over every `MachO` binary contained in this `MultiArch` container
-pub struct MachOIterator<'a> {
+/// Iterator over every entry contained in this `MultiArch` container
+pub struct MultiArchEntryIterator<'a> {
```

Comment:
ditto name here; I'm glad you named this after the Variant above, so if that changes, we should keep it consistent like you've done here